core/src/main/java/hudson/model/RunParameterDefinition.java

```diff
@@ -152,9 +153,9 @@ public String getHelpFile() {
         public ParameterDefinition newInstance(StaplerRequest req, JSONObject formData) throws FormException {
             return req.bindJSON(RunParameterDefinition.class, formData);
         }
-        
-        public AutoCompletionCandidates doAutoCompleteProjectName(@QueryParameter String value) {
-            return AutoCompletionCandidates.ofJobNames(Job.class, value, null, Jenkins.get());
+
+        public AutoCompletionCandidates doAutoCompleteProjectName(@QueryParameter String value, @AncestorInPath ItemGroup context) {
+            return AutoCompletionCandidates.ofJobNames(value, context);
```

Comment:
Does it not work to just do `AutoCompletionCandidates.ofJobNames(Job.class, value, context);` and leverage existing code?